### ISSUE TITLE
Add more functionality to plotting stage

### ISSUE THREAD

**Issue by bistapf** [HEP-FCC/FCCAnalyses#420]
Additional functionalities to be added to the plotting stage scripts: 

- [ ] Option to normalize the histograms by to unit, requested in https://github.com/HEP-FCC/FCCAnalyses/issues/414
- [ ] Option to add the overlow to the last bin
- [ ] Option to add a ratio panel (usually signal/background) below the main plot

All these functionalities exist in https://github.com/bistapf/FCCAnalyses/blob/FCChh_HH_analyses/run/bb2lMET_analysis/draw_plots.py (which is a mess) and could be copy/pasted.

**Comment by bistapf** [comment]
Another "nice to have" feature would be support for logarithmic x-axis

**Comment by thisismeamir** [comment]
Thanks for filing this, and for the pointer to the reference implementation. The three requested features (normalization, overflow, ratio panel) are well-scoped and clearly needed. Rather than copy-pasting them into the existing `do_plots.py`, I'd like to use this as an opportunity to also address some structural issues in the current module that would otherwise make each of these additions harder to maintain. Here's the plan.

The current `do_plots.py` has a few architectural patterns that accumulate technical debt quickly:

- **Duplicated histogram loading and legend construction** between the staged and histmaker flows (`load_hists` vs. `mapHistosFromHistmaker`, `runPlots` vs. `runPlotsHistmaker`). Any new feature added to one must be manually mirrored in the other.
- **`draw_plot` is a ~300-line monolithic function** that handles canvas setup, axis ranges, stacking, stat uncertainties, all text annotations, and the yields summary page in one body. Adding a ratio pad into this function safely is difficult without introducing regressions.
- **Configuration is a raw string-keyed dict** passed through every call chain, making the interface fragile and hard to validate.
- **Two analysis modes diverge immediately** via `sys.exit()` after a type check, preventing shared infrastructure.

The refactor I'm planning will:

1. **Introduce a `PlotConfig` dataclass** – a typed, validated configuration object constructed once from the user script + CLI args. All config keys become typed attributes with defaults. New options like `normalize`, `add_overflow`, and `ratio_panel` are added here with sensible defaults so existing scripts are fully backward-compatible.

2. **Unify histogram loading** into a single `HistogramLoader` class with a common interface. Both analysis modes (staged and histmaker) produce `dict[str, TH1]` outputs; only the file discovery and scale logic differ. Overflow merging and normalization are applied here, once, for both modes.

3. **Decompose `draw_plot` into a `PlotDrawer` class** with clearly separated methods for canvas setup, stack drawing, stat uncertainty, annotation, and output. A ratio pad becomes a dedicated method (`draw_ratio`) and a dedicated `RatioPanel` class that computes S/B, S/√B, or data/MC ratios — each independently testable.

4. **Split the module into submodules** (`config.py`, `loader.py`, `drawer.py`, `yields.py`, `utils.py`) under `python/FCCAnalyses/plot/`, keeping the public entry point `do_plots()` unchanged from the user's perspective.

Existing plot scripts will require **no changes**. All new config options default to the current behavior (`normalize=False`, `add_overflow=False`, `ratio_panel=None`). The `do_plots()` entry point and the user-facing script API (attribute names like `intLumi`, `stacksig`, `formats`, etc.) remain unchanged.

I plan to open a draft PR with the refactored structure first (no behavior change, all tests passing), then layer in the three new features on top so the diff is reviewable in stages. Happy to discuss the design before starting if there are preferences on the module layout or the ratio panel API.
